### PR TITLE
feat: UI flow for event management

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview

- Implement ui for event management

## Demo

Dec-26-2023 17-42-42

### CODE REVIEW

**Comment by mthinh** [comment]
@sieuhuflit Why dont we use svg for icon ?

File: lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart [hunk 1]
```diff
@@ -63,6 +64,11 @@ class _GuestEventDetailView extends StatelessWidget {
 
           final isAttending =
               EventUtils.isAttending(event: event, userId: userId);
+          final isOwnEvent =
+              EventUtils.isOwnEvent(event: event, userId: userId);
+          if (isOwnEvent) {
+            return const PostHostEventDetailView();
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart:70]: `Post` as a prefix is not necessary in this context

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    padding: EdgeInsets.only(top: 6.h),
+                  ),
+                  SliverPadding(
+                    padding: EdgeInsets.symmetric(horizontal: Spacing.smMedium),
+                    sliver: SliverToBoxAdapter(
+                      child: HostEventBasicInfoCard(event: event),
+                    ),
+                  ),
+                  SliverPadding(
+                    padding: EdgeInsets.only(top: 36.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart:55]: please use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                physics: const BouncingScrollPhysics(),
+                slivers: [
+                  SliverPersistentHeader(
+                    pinned: true,
+                    delegate: PostGuestEventAnimatedAppBar(
+                      event: event,
+                    ),
+                  ),
+                  SliverPadding(
+                    padding: EdgeInsets.only(top: 6.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart:46]: pls use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    padding: EdgeInsets.only(top: 36.h),
+                  ),
+                  SliverPadding(
+                    padding: EdgeInsets.symmetric(horizontal: Spacing.smMedium),
+                    sliver: HostEventDetailConfigGrid(
+                      event: event,
+                    ),
+                  ),
+                  SliverPadding(
+                    padding: EdgeInsets.only(top: 42.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/views/post_host_event_detail_view.dart:64]: please use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart (hunk 1 above)
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart:70]: should create another event page for host. Should not combine all views in one page.

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_basic_info_card.dart [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+import 'package:app/injection/register_module.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:app/theme/typo.dart';
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:dartz/dartz.dart';
+import 'package:duration/duration.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class HostEventBasicInfoCard extends StatelessWidget {
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_basic_info_card.dart:24]: lets move these widgets to another folder 
/pages/event/host_event_detail_page/widgets

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: Assets.images.icUsersAdd.provider(),
+            ),
+          ),
+        ),
+      ),
+      EventConfigGridViewModel(
+        title: t.event.configuration.tickets,
+        subTitle: '4 ticket types \n\$12.9K total sales',
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:43]: remove hard code

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+
+  final Event event;
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final List<EventConfigGridViewModel?> listData = [
+      EventConfigGridViewModel(
+        title: t.event.configuration.invite,
+        subTitle: '12/48 confirmed',
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:24]: remove hardcoded

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: Assets.images.icTicketGradient.provider(),
+            ),
+          ),
+        ),
+      ),
+      EventConfigGridViewModel(
+        title: t.event.configuration.checkIn,
+        subTitle: '1.2k checked in',
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:57]: remove hardcoded

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+        showProgressBar: true,
+        progressBarColors: [const Color(0xFFF9D3BC), const Color(0xFFF29A68)],
+        icon: Container(
+          width: 24.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:61]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: Assets.images.icCheckInGradient.provider(),
+            ),
+          ),
+        ),
+      ),
+      EventConfigGridViewModel(
+        title: t.event.configuration.coHosts,
+        subTitle: '',
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:73]: why empty ?

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+        icon: Container(
+          width: 24.w,
+          height: 24.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:76]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+            ),
+          ),
+        ),
+      ),
+      EventConfigGridViewModel(
+        title: t.event.configuration.controlPanel,
+        subTitle: t.event.configuration.controlPanelDescription,
+        icon: Container(
+          width: 24.w,
+          height: 24.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:90]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+            ),
+          ),
+        ),
+      ),
+      EventConfigGridViewModel(
+        title: t.event.configuration.dashboard,
+        subTitle: t.event.configuration.dashboardDescription,
+        icon: Container(
+          width: 24.w,
+          height: 24.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:104]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+        mainAxisSpacing: 10,
+        childAspectRatio: 0.95,
+      ),
+      delegate: SliverChildBuilderDelegate(
+        (BuildContext context, int index) {
+          return GridItemWidget(
+            item: listData[index],
+          );
+        },
+        childCount: 6, // Total number of items
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:127]: define const inside the widget will be fine

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+  const GridItemWidget({
+    super.key,
+    required this.item,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    return Container(
+      padding: const EdgeInsets.only(top: 1, left: 1, right: 1, bottom: 6),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:145]: use .w and Spacing support 6 also

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+      decoration: ShapeDecoration(
+        color: colorScheme.secondaryContainer,
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(16),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:149]: use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+        ),
+      ),
+      child: Container(
+        padding: const EdgeInsets.all(15),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:153]: use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 18]
```diff
@@ -0,0 +1,246 @@
...
+        clipBehavior: Clip.antiAlias,
+        decoration: ShapeDecoration(
+          gradient: LinearGradient(
+            begin: const Alignment(0.00, -1.00),
+            end: const Alignment(0, 1),
+            colors: [colorScheme.secondaryContainer, Colors.black],
+          ),
+          shape: RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(15),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:162]: use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 19]
```diff
@@ -0,0 +1,246 @@
...
+                    child: Text(
+                      item?.title ?? '',
+                      style: Typo.small.copyWith(
+                        fontFamily: FontFamily.nohemiVariable,
+                        color: Colors.white,
+                        fontWeight: FontWeight.w700,
+                      ),
+                    ),
+                  ),
+                  // const SizedBox(height: 6),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:188]: remove unused

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 20]
```diff
@@ -0,0 +1,246 @@
...
+                  item!.showProgressBar == true
+                      ? Padding(
+                          padding: EdgeInsets.symmetric(vertical: 6.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:191]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 21]
```diff
@@ -0,0 +1,246 @@
...
+                          child: Container(
+                            width: 79,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:193]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 22]
```diff
@@ -0,0 +1,246 @@
...
+                            height: 2,
+                            decoration: ShapeDecoration(
+                              color: colorScheme.outline,
+                              shape: RoundedRectangleBorder(
+                                borderRadius: BorderRadius.circular(100),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:198]: why radius 100 ?

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 23]
```diff
@@ -0,0 +1,246 @@
...
+                              ),
+                            ),
+                            child: Stack(
+                              children: [
+                                Positioned(
+                                  left: -0,
+                                  top: 0,
+                                  child: Container(
+                                    width: 57,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:207]: use .w,

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 24]
```diff
@@ -0,0 +1,246 @@
...
+                                    height: 2,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:208]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart [hunk 25]
```diff
@@ -0,0 +1,246 @@
...
+                            ),
+                          ),
+                        )
+                      : const SizedBox(),
+                  SizedBox(height: 2.h),
+                  SizedBox(
+                    child: Text(
+                      item?.subTitle ?? '',
+                      style: Typo.xSmall.copyWith(
+                        fontSize: 9,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:232]: use Typo

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_location.dart [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+      width: double.infinity,
+      height: 144.w,
+      child: Row(
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          SizedBox(
+            width: 144.w,
+            child: ClipRRect(
+              borderRadius: BorderRadius.only(
+                topLeft: Radius.circular(15.sp),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_location.dart:37]: why .sp used for radius ? use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_location.dart [hunk 27]
```diff
@@ -0,0 +1,107 @@
...
+          Expanded(
+            child: Container(
+              padding: EdgeInsets.symmetric(
+                vertical: Spacing.smMedium,
+                horizontal: Spacing.small,
+              ),
+              decoration: BoxDecoration(
+                color: colorScheme.surfaceVariant,
+                borderRadius: BorderRadius.only(
+                  topRight: Radius.circular(15.sp),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_location.dart:74]: why .sp used for radius ? use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 28]
```diff
@@ -0,0 +1,268 @@
...
+
+  final Event event;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final t = Translations.of(context);
+    return Container(
+      width: double.infinity,
+      height: 229.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:27]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 29]
```diff
@@ -0,0 +1,268 @@
...
+      padding: const EdgeInsets.all(24),
+      clipBehavior: Clip.antiAlias,
+      decoration: ShapeDecoration(
+        shape: RoundedRectangleBorder(
+          side: BorderSide(
+            width: 2.w,
+            color: colorScheme.outline,
+          ),
+          borderRadius: BorderRadius.circular(15),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:36]: use LemonRadius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 30]
```diff
@@ -0,0 +1,268 @@
...
+              SizedBox(
+                width: double.infinity,
+                child: Row(
+                  mainAxisSize: MainAxisSize.min,
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    Container(
+                      width: 72.w,
+                      height: 72.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:55]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 31]
```diff
@@ -0,0 +1,268 @@
...
+                      clipBehavior: Clip.antiAlias,
+                      decoration: ShapeDecoration(
+                        color: Colors.white.withOpacity(0.05999999865889549),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:58]: can we rounded to 0.06 instead?

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 32]
```diff
@@ -0,0 +1,268 @@
...
+                        shape: RoundedRectangleBorder(
+                          borderRadius: BorderRadius.circular(9),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:60]: use lemon radius

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 33]
```diff
@@ -0,0 +1,268 @@
...
+                        ),
+                      ),
+                    ),
+                    SizedBox(width: 18.w),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:64]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 34]
```diff
@@ -0,0 +1,268 @@
...
+                      child: Column(
+                        mainAxisSize: MainAxisSize.min,
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Container(
+                            width: 60.w,
+                            height: 18.h,
+                            padding: EdgeInsets.symmetric(
+                              horizontal: 36.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:75]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 35]
```diff
@@ -0,0 +1,268 @@
...
+                    ),
+                    SizedBox(width: 18.w),
+                    Expanded(
+                      child: Column(
+                        mainAxisSize: MainAxisSize.min,
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        crossAxisAlignment: CrossAxisAlignment.start,
+                        children: [
+                          Container(
+                            width: 60.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:72]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 36]
```diff
@@ -0,0 +1,268 @@
...
+                            ),
+                            clipBehavior: Clip.antiAlias,
+                            decoration: ShapeDecoration(
+                              color: LemonColor.white06,
+                              shape: RoundedRectangleBorder(
+                                borderRadius: BorderRadius.circular(3.r),
+                              ),
+                            ),
+                          ),
+                          SizedBox(height: 9.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:86]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 37]
```diff
@@ -0,0 +1,268 @@
...
+                          Container(
+                            width: double.infinity,
+                            height: 27.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:89]: use Sizing or .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 38]
```diff
@@ -0,0 +1,268 @@
...
+                            clipBehavior: Clip.antiAlias,
+                            decoration: const BoxDecoration(),
+                            child: Column(
+                              mainAxisSize: MainAxisSize.min,
+                              mainAxisAlignment: MainAxisAlignment.start,
+                              crossAxisAlignment: CrossAxisAlignment.start,
+                              children: [
+                                Container(
+                                  width: double.infinity,
+                                  height: 12.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:99]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 39]
```diff
@@ -0,0 +1,268 @@
...
+                                  padding: EdgeInsets.symmetric(
+                                    horizontal: 36.w,
+                                    vertical: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:102]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 40]
```diff
@@ -0,0 +1,268 @@
...
+                                    color: LemonColor.white06,
+                                    shape: RoundedRectangleBorder(
+                                      borderRadius: BorderRadius.circular(3.r),
+                                    ),
+                                  ),
+                                ),
+                                SizedBox(height: 3.h),
+                                Container(
+                                  width: 160.w,
+                                  height: 12.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:115]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 41]
```diff
@@ -0,0 +1,268 @@
...
+                                  padding: EdgeInsets.symmetric(
+                                    horizontal: 36.w,
+                                    vertical: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:118]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 42]
```diff
@@ -0,0 +1,268 @@
...
+                                      borderRadius: BorderRadius.circular(3.r),
+                                    ),
+                                  ),
+                                ),
+                                SizedBox(height: 3.h),
+                                Container(
+                                  width: 160.w,
+                                  height: 12.h,
+                                  padding: EdgeInsets.symmetric(
+                                    horizontal: 36.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:117]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 43]
```diff
@@ -0,0 +1,268 @@
...
+                width: double.infinity,
+                height: 85.h,
+                child: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    SizedBox(
+                      width: double.infinity,
+                      height: 28,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:148]: use Sizing or .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 44]
```diff
@@ -0,0 +1,268 @@
...
+                        ],
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              SizedBox(height: 20.h),
+              SizedBox(
+                width: double.infinity,
+                height: 85.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:140]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 45]
```diff
@@ -0,0 +1,268 @@
...
+                            width: double.infinity,
+                            clipBehavior: Clip.antiAlias,
+                            decoration: const BoxDecoration(),
+                            child: Row(
+                              mainAxisSize: MainAxisSize.min,
+                              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                              crossAxisAlignment: CrossAxisAlignment.center,
+                              children: [
+                                Container(
+                                  width: 60.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:164]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 46]
```diff
@@ -0,0 +1,268 @@
...
+                                  height: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:165]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 47]
```diff
@@ -0,0 +1,268 @@
...
+                                  padding: EdgeInsets.symmetric(
+                                    horizontal: 36.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:167]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 48]
```diff
@@ -0,0 +1,268 @@
...
+                                    vertical: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:168]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 49]
```diff
@@ -0,0 +1,268 @@
...
+                                  ),
+                                  clipBehavior: Clip.antiAlias,
+                                  decoration: ShapeDecoration(
+                                    color: LemonColor.white06,
+                                    shape: RoundedRectangleBorder(
+                                      borderRadius: BorderRadius.circular(3.r),
+                                    ),
+                                  ),
+                                ),
+                                SizedBox(width: 12.w),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:178]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 50]
```diff
@@ -0,0 +1,268 @@
...
+                                Container(
+                                  width: 60.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:180]: use Sizing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 51]
```diff
@@ -0,0 +1,268 @@
...
+                                  height: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:181]: use Sizing or .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 52]
```diff
@@ -0,0 +1,268 @@
...
+                                  padding: EdgeInsets.symmetric(
+                                    horizontal: 36.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:183]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 53]
```diff
@@ -0,0 +1,268 @@
...
+                                    vertical: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:184]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 54]
```diff
@@ -0,0 +1,268 @@
...
+                                    ),
+                                  ),
+                                ),
+                              ],
+                            ),
+                          ),
+                        ],
+                      ),
+                    ),
+                    SizedBox(height: 15.h),
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:200]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 55]
```diff
@@ -0,0 +1,268 @@
...
+                    Container(
+                      width: double.infinity,
+                      height: 42.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:203]: use .w

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 56]
```diff
@@ -0,0 +1,268 @@
...
+                      padding: EdgeInsets.symmetric(
+                        horizontal: 36.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:205]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 57]
```diff
@@ -0,0 +1,268 @@
...
+                        vertical: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:206]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 58]
```diff
@@ -0,0 +1,268 @@
...
+                SizedBox(
+                  height: 9.h,
+                ),
+                Text(
+                  t.event.rewardsDescription,
+                  textAlign: TextAlign.center,
+                  style: Typo.small.copyWith(color: colorScheme.onSecondary),
+                ),
+                SizedBox(
+                  height: 18.h,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:245]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 59]
```diff
@@ -0,0 +1,268 @@
...
+                  width: 108.w,
+                  label: t.event.addPoap,
+                  trailing: ThemeSvgIcon(
+                    builder: (filter) =>
+                        Assets.icons.icSendMessage.svg(colorFilter: filter),
+                  ),
+                  mode: GradientButtonMode.lavenderMode,
+                  radius: BorderRadius.circular(LemonRadius.normal),
+                  padding: EdgeInsets.symmetric(
+                    horizontal: 12.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:257]: use Spacing

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart [hunk 60]
```diff
@@ -0,0 +1,268 @@
...
+                    vertical: 9.w,
```
Comment (mthinh) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_poap_reward_card.dart:258]: use Spacing

**Review by mthinh — CHANGES_REQUESTED** [review]
Left some comments

File: lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart (hunk 1 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/event/guest_event_detail_page/guest_event_detail_page.dart:70]: Thanks, this super helpful

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart (hunk 10 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:73]: co-hosts detail UI will customize later when implement co-host part

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart (hunk 22 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:198]: it's design make it 100, I'm adjust to 100.r

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart (hunk 25 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/host_event_detail_config_grid.dart:232]: Already use, but this one special case, Typo doesn't contain this fontSize 9, this font size rarely use

**Review by mthinh — APPROVED** [review]

**Pull request merged**